### PR TITLE
Workflowtemp2 (#397)

### PR DESCRIPTION
**Pull request by matthewchungcodefresh**
* Add files via upload

* Update README.md

### CODE REVIEW

File: examples/image-test-workflow/versions/0.0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# JIRA-SCAN-WORKFLOW
+
+## Summary
+
+This Workflow Template is used to create a CI/CD pipeline that creates and tests an image through a service container
+
+## Templates
+
+1. [image-test-template](https://github.com/codefresh-io/argo-hub/blob/main/workflows/image-test-template/versions/0.0.1/docs/image-test-template.md)
```
Comment (lrochette) [inline examples/image-test-workflow/versions/0.0.1/README.md:9]: path needs to be changed, it now examples not workflows

File: examples/image-test-workflow/versions/0.0.1/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+## Templates
+
+1. [image-test-template](https://github.com/codefresh-io/argo-hub/blob/main/workflows/image-test-template/versions/0.0.1/docs/image-test-template.md) 
+
+## Security
+
+Minimal required permissions
+
+[Full rbac permissions list](https://github.com/codefresh-io/argo-hub/blob/main/workflows/image-test-template/versions/0.0.1/rbac.yaml)
```
Comment (lrochette) [inline examples/image-test-workflow/versions/0.0.1/README.md:15]: path needs to be changed, it now examples not workflows

File: examples/image-test-workflow/versions/0.0.1/docs/image-test-template.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
+# image-test-template
+
+## Summary
+This Workflow Template is used to create a pipeline that takes a source repository and creates an image through Kaniko. Using that image, a service container is set up and the image is tested using the service container. This example runs nodejs on a mysql server. Template based off https://codefresh.io/docs/docs/yaml-examples/examples/integration-tests-with-mysql/#the-example-nodejs-project.
+
+## Inputs/Outputs
+
+### Inputs
```
Comment (lrochette) [inline examples/image-test-workflow/versions/0.0.1/docs/image-test-template.md:8]: the lis of parameters does not match the list from workflowTemplate.yaml

Comment (lrochette) [reply examples/image-test-workflow/versions/0.0.1/docs/image-test-template.md:8]: I think this list is  more accurate than the template one to be honest

File: examples/image-test-workflow/versions/0.0.1/docs/image-test-template.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+### Secrets
+* MYSQL Secret - in order for this template to work a secret named `mysql-secret` must exist with mysql login information.
+```
+    apiVersion: v1
+    kind: Secret
+    metadata:
+        name: mysql-secret
+    type: Opaque
+    data:
+        root-password: echo -n 'mysql root password' | base64
```
Comment (lrochette) [inline examples/image-test-workflow/versions/0.0.1/docs/image-test-template.md:37]: if we create a secret with all the data, why do we need to pass individual values to the main template

          - name: MYSQL_SECRET
            default: mysql-secret
          - name: MYSQL_ROOT_PASSWORD
            default: root-password
          - name: MYSQL_USER
            default: user
          - name: MYSQL_PASSWORD
            default: password

only the first one is requried

File: examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            default: docker-config  # the name of the registry-creds secret you created
+          # Create MYSQL secret and apply it to your cluster under the runtime namespace
+          #   kubectl create secret generic mysql-secret \
+          #   --from-literal=root-password=$MYSQL_ROOT_PASSWORD --dry-run=client \
+          #   --from-literal=user=$MYSQL_USER --dry-run=client \
+          #   --from-literal=password=$MYSQL_PASSWORD --dry-run=client \
+          #   --save-config -o yaml | kubectl apply -f - -n $NAMESPACE
+          - name: MYSQL_SECRET
+            default: mysql-secret
+          - name: MYSQL_ROOT_PASSWORD
```
Comment (lrochette) [inline examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml:46]: those are not required at the top template as they are values from the secret

File: examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+            default: root-password
+          - name: MYSQL_USER
+            default: user
+          - name: MYSQL_PASSWORD
+            default: password
+          - name: MYSQL_DATABASE
+          - name: MYSQL_HOST
+          - name: REPO_URL
```
Comment (lrochette) [inline examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml:54]: Add default value of codefreshdemo/cf-example-unit-tests-with-composition

File: examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+          - name: REVISION
+            default: main
```
Comment (lrochette) [inline examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml:56]: default branch is master for the repo above

File: examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml (hunk 5 above)
Comment (lrochette) [reply examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml:46]: I see.
those are the keys to extract value  from the secret
just keep them hardcoded at line 147, 152 and 157 to limit the numbers of parameters

**Review by lrochette — CHANGES_REQUESTED** [review]

**Review by lrochette — CHANGES_REQUESTED** [review]
when I try to run it I get error:
templates.build inputs.parameters.REPO was not supplied 

EDIT: strange I cannot find reference to that code

File: examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+  annotations:
+    argo-hub/version: '0.0.1'
+    argo-hub/description: 'Workflow template for a deployment that builds and tests an image through service containers'
+    argo-hub/categories: 'argo'
+    argo-hub/license: 'MIT'
+    argo-hub/owner_name: 'Matthew Chung'
+    argo-hub/owner_email: 'matthew.chung@codefresh.io'
+    argo-hub/owner_avatar: 'https://github.com/matthewchungcodefresh.png'
+    argo-hub/owner_url: 'https://github.com/matthewchungcodefresh'
+    argo-hub/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/examples/image-test-workflow/assets/icon.svg"
```
Comment (eti-codefresh) [inline examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml:14]: need to be change to this "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/examples/image-test-workflow/assets/icon.svg"

File: examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+          - key: .dockerconfigjson
+            path: config.json
+        secretName: '{{ inputs.parameters.DOCKER_CONFIG_SECRET }}'
+  entrypoint: main
+  templates:
+    - name: main
+      metadata:
+        annotations:
+          argo-hub-template/description: 'Builds a docker image using Kaniko and tests the image through service containers'
+          argo-hub-template/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/examples/image-test-workflow/assets/icon.svg"
```
Comment (eti-codefresh) [inline examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml:30]: need to be change to this "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/examples/image-test-workflow/assets/icon.svg"

File: examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+            arguments:
+              parameters:
+                - name: IMAGE
+                  value: '{{ inputs.parameters.IMAGE_NAME }}:{{ inputs.parameters.TAG }}'
+
+    - name: mysql-service
+      metadata:
+        annotations:
+          argo-hub-template/description: 'Build service container for MYSQL'
+          argo-hub-template/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/examples/image-test-workflow/assets/icon.svg"
```
Comment (eti-codefresh) [inline examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml:119]: need to be change to this "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/examples/image-test-workflow/assets/icon.svg"

File: examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+          - name: MYSQL_DATABASE
+            value: '{{ inputs.parameters.MYSQL_DATABASE }}'
+          - name: MYSQL_HOST
+            value: '{{ inputs.parameters.MYSQL_HOST }}'
+
+    - name: mysql-client
+      metadata:
+        annotations:
+          argo-hub-template/description: 'Initialize service container and test image'
+          argo-hub-template/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/examples/image-test-workflow/assets/icon.svg"
```
Comment (eti-codefresh) [inline examples/image-test-workflow/versions/0.0.1/workflowTemplate.yaml:159]: need to be change to this "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/examples/image-test-workflow/assets/icon.svg"

**Review by eti-codefresh — CHANGES_REQUESTED** [review]

**Review by lrochette — APPROVED** [review]

**Comment by lrochette** [comment]
Too much junk. Closing and deleting branch

**Pull request closed without merging**